GitCommands/AsyncLoader.cs

```diff
@@ -115,48 +95,54 @@ void Load()
                     loadContent(token);
                 }
             }
-
-            void Continuation(Task task)
+            catch (Exception e)
             {
-                if (task.IsFaulted)
+                if (e is OperationCanceledException && token.IsCancellationRequested)
                 {
-                    Debug.Assert(task.Exception != null, "Faulted task should have a non-null exception");
+                    return;
+                }
 
-                    foreach (var e in task.Exception.InnerExceptions)
-                    {
-                        if (!OnLoadingError(e))
-                        {
-                            throw e;
-                        }
-                    }
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
📝 This requested change would not preserve the original semantics. If the original action faults (as opposed to being cancelled), the error handling callback cannot be independently cancelled.

Reply:
Perhaps you are right, to confirm that I would have to know details of how TaskContinuationOptions.NotOnCanceled is implemented. The proposed by you implementation can lead to a situation when exception handling is done when the task is canceled which in case of AsyncLoader  often translates to: the caller form is disposed. Thus even if it means we are changing the bahaviour here, I am for doing such a change.

This a bight side of the verbosity of JTF.

Reply:
> The proposed by you implementation can lead to a situation when exception handling is done when the task is canceled which in case of AsyncLoader often translates to: the caller form is disposed.

I don't see how that is the case. In the event of a cancellation exception, the block four lines above should serve as an early return. This code should only execute if the call to `loadContent` throws an exception which is _not_ simply cancellation.

Reply:
The sequence is:
1. `loadContent` throws an exeption (let's say NRE)
2. We enter the catch block and awaiting on switching to the main thread.
3. While we are awaiting in the background thread, the user closes the caller form, which gets disposed.
4. After 3 is done, the message loop on the main thread gets next item which processes OnLoadingError.

Reply:
Yes, that is the expected behavior here (at least in the sense it is how I intended to implement this PR). `loadContent` failed (as opposed to being cancelled) so `OnLoadingError` will be called. There is no disposal protection for OnLoadingError before or after my change.

Reply:
As I wrote: Thus even if it means we are changing the bahaviour here, I am for doing such a change.
Do you mind doing so?

Reply:
It's not as easy to implement it this way as expected, because `AsyncLoader` isn't associated with a specific control (i.e. you can't use `control.SwitchToMainThreadAsync`). Also, it seems reasonable for the caller to want generalized error handling support, so I'm not sure refactoring everything to drop error handling is a desirable at that level.

Reply:
Are you sure you want
 ```c# 
throw e;
``` 
and not
 ```c# 
throw;
```

Reply:
@vbjay The code in this pull request uses `throw` throughout the file